actionview/lib/action_view/railtie.rb

```diff
@@ -33,6 +33,8 @@ class Railtie < Rails::Railtie # :nodoc:
         if app.config.action_view.cache_template_loading.nil?
           ActionView::Resolver.caching = app.config.cache_classes
         end
+
+        PartialRenderer.collection_cache = Rails.cache
```

Comment:
I ditched the initializer and used `Rails.try(:cache)` in the `collection_cache` definition. I still changed it to a `mattr_accessor`, so Action View users outside of Rails can switch cache store if the in memory one isn't a fit.